### PR TITLE
Md 11/loader

### PR DESCRIPTION
**Pull request by Danny-Mus**
Loader interceptor for all api calls

2 thumbsup plz

### CODE REVIEW

File: front-dex/src/app/services/loader/loader.service.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+@Injectable()
+export class LoaderService {
+    isLoading = new Subject<boolean>();
+    show() {
```
Comment (DaanEKR) [inline front-dex/src/app/services/loader/loader.service.ts]: add function type

File: front-dex/src/app/services/loader/loader.service.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+        this.isLoading.next(true);
+    }
+    hide() {
```
Comment (DaanEKR) [inline front-dex/src/app/services/loader/loader.service.ts]: add function type

File: front-dex/src/app/components/loader/loader.component.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<div *ngIf="isLoading | async" class="overlay">
```
Comment (DaanEKR) [inline front-dex/src/app/components/loader/loader.component.html:1]: Do you have to use a div?

File: front-dex/src/app/components/loader/loader.component.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
+@import '../../../assets/scss/main.scss';
+
+.overlay {
+    position: fixed;
+    display: block;
+    width: 100%;
+    height: 100%;
+    top: 0;
+    left: 0;
+    background-color: rgba(211, 211, 211, 0.8);
```
Comment (kmensink95) [inline front-dex/src/app/components/loader/loader.component.scss]: Use a variable

File: front-dex/src/app/components/loader/loader.component.scss [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        top: 50%;
+        left: 50%;
+        transform: translate(-120%,-50%) scaleX(-1);
+        background: url(../../../assets/icons/ruby.gif);
+        background-size: 300px;
+
+        &--pokemon1 {
+            background: url(../../../assets/icons/jolteon_walking.gif);
+            background-size: 250px;
+            background-repeat: no-repeat;
```
Comment (kmensink95) [inline front-dex/src/app/components/loader/loader.component.scss]: no repeat can behind the background url and maybe the size too?

File: front-dex/src/app/components/loader/loader.component.scss [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            transform: translate(-35%, -28%);
+        }
+
+        &--pokemon2 {
+            background: url(../../../assets/icons/glaceon_walking.gif);
+            background-size: 230px;
+            background-repeat: no-repeat;
```
Comment (kmensink95) [inline front-dex/src/app/components/loader/loader.component.scss]: Same story

**Pull request merged**